[TC-770] - Consolidate Analytics

carlosthe19916 opened the pull request:
We need to put the analytics code at one single point and avoid spreading of "event names" as much as we can.

- [x] Log the search entries on the Home Page + Log the search at the "Search results" page
- [x] Track the selected tabs on the "Search results" page
- [x] What SBOMs are being downloaded + number of times the download link is clicked (this has been implemented before this PR  by Jens)
- [x] Count number of successful and unsuccessful sboms being scanned (This was done before this PR by Jens)
- [ ] Log what filters are used while searching in the "Search results" page

The idea is to put all "event keys" in one single place so it is easier for us to keep good naming convention. I suggest also following good practices described at https://segment.com/docs/getting-started/implementation-guide/#event-anatomy-and-naming-standards 

```
Segment recommends the best practice of using an “Object Action” (Noun Verb)
naming convention for all Track events (for example, Menu Clicked)
```

ctron commented on spog/ui/src/pages/search.rs:
```diff
@@ -25,6 +28,17 @@ pub enum TabIndex {
     Packages,
 }
 
+impl std::fmt::Display for TabIndex {
```
Just a comment, there is a crate call `strum` which can do that for you. But it's ok to merge this anyway.

carlosthe19916 replied:
@ctron Thanks for the advice. I didn't realize we are using `strum`. I've just applied it on the last commit! Thanks

carlosthe19916 commented:
@ctron I added some analytics on my own, and also renamed the current "Even Keys" that already existed for scanning SBOMs, my idea was to make the event names more comprehensible, feel free to comment on those changes and suggest better names or simply reject the name changes :)

I still need to cover the last task "track which filters are used", but I changed this PR to "Ready to be reviewed" as it would be better to have a review and perhaps move on with what we have. You or I can create another PR for covering the filters later on.

Let me know what you think.

![Uploading Screenshot from 2023-12-06 09-32-06.png…]()

ctron commented:
Naming is hard, I wasn't sure what was expected. So I went with a very, very simple approach. Looks better to me now!
